src/effects/VST3/internal/ParameterQueuePool.h

```diff
@@ -0,0 +1,49 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file ParameterQueuePool.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include <functional>
+#include <pluginterfaces/base/smartpointer.h>
+#include <pluginterfaces/vst/vsttypes.h>
+
+#include "ParameterValueQueue.h"
+
+namespace internal
+{
+
+   class ParameterValueQueue;
+
+   //Helper object, saves some cycles by avoiding
+   //temporary objects reallocation
+   class ParameterQueuePool final
+   {
+      std::vector<Steinberg::IPtr<ParameterValueQueue>> mPool;
+   public:
+
+      using ParameterValueQueuePtr = std::unique_ptr<
+         ParameterValueQueue,
+         std::function<void(ParameterValueQueue*)>
```

Comment:
The deleter in the .cpp file is always the same lambda, with an empty capture, so you could declare a zero-sized deleter class here with an `operator ()`, and define it in the .cpp file, and make this unique_ptr just one pointer wide.

Reply:
Good point